apps/prairielearn/src/tests/activeAccessRestriction.test.ts

```diff
@@ -186,6 +203,27 @@ describe('Exam and homework assessment with active access restriction', function
     assert.equal(results.rows[0].open, false);
   });
 
+  step('access question with existing variant when exam is closed', async () => {
+    const response = await helperClient.fetchCheerio(context.examQuestionUrl, {
+      headers,
+    });
+    assert.isTrue(response.ok);
+
+    // There should be no save or grade buttons
+    assert.lengthOf(response.$('button.question-save'), 0);
+    assert.lengthOf(response.$('button.question-grade'), 0);
+  });
+
+  step('access question without existing variant when exam is closed', async () => {
+    const response = await helperClient.fetchCheerio(context.examQuestionWithoutVariantUrl, {
+      headers,
+    });
+    assert.equal(response.status, 403);
+
+    const result = response.$('div.card-body:contains(' + VARIANT_FORBIDDEN_STRING + ')');
```

Comment:
Nit, we generally prefer to use templates literals; you can make this change elsewhere as well:

```suggestion
    const result = response.$(`div.card-body:contains(${VARIANT_FORBIDDEN_STRING })`);
```